src/config.rs

```diff
@@ -509,29 +528,39 @@ pub fn get_config_dir() -> Result<(PathBuf, PathSource)> {
 ///
 /// Note that this function does not verify whether the file at that location
 /// exists, or is a file.
-pub fn get_config_path() -> Result<(PathBuf, PathSource)> {
+pub fn get_default_config_path() -> Result<PathWithSource> {
     let (config_dir, source) = get_config_dir()?;
     let config_file_path = config_dir.join(CONFIG_FILE_NAME);
-    Ok((config_file_path, source))
+    Ok(PathWithSource {
+        path: config_file_path,
+        source,
+    })
 }
 
 /// Create default config file.
-pub fn make_default_config() -> Result<PathBuf> {
-    let (config_dir, _) = get_config_dir()?;
-
-    // Ensure that config directory exists
-    if config_dir.exists() {
-        ensure!(
-            config_dir.is_dir(),
-            "Config directory could not be created: {} already exists but is not a directory",
-            config_dir.to_string_lossy(),
-        );
+/// path: Can be specified to create the config in that path instead of
+/// the default path.
+pub fn make_default_config(path: Option<&Path>) -> Result<PathBuf> {
+    let config_file_path = if let Some(p) = path {
+        p.into()
     } else {
-        fs::create_dir_all(&config_dir).context("Could not create config directory")?;
-    }
+        let (config_dir, _) = get_config_dir()?;
+
+        // Ensure that config directory exists
```

Comment:
Ah, so here we create the parent directories only if the default path is used. I think I agree with the decision - can we document it by adding a test?